src/features/hotspots/setup/HotspotSetupAddTxnScreen.tsx

```diff
@@ -0,0 +1,168 @@
+import { useNavigation } from '@react-navigation/native'
+import React, { useEffect, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import BackScreen from '../../../components/BackScreen'
+import Box from '../../../components/Box'
+import Button from '../../../components/Button'
+import Text from '../../../components/Text'
+import { useConnectedHotspotContext } from '../../../providers/ConnectedHotspotProvider'
+import { RootState } from '../../../store/rootReducer'
+import { currencyFormat } from '../../../utils/currency'
+import { calculatePaymentTxnFee } from '../../../utils/transactions'
+import usePermissionManager from '../../../utils/usePermissionManager'
+import { HotspotSetupNavigationProp } from './hotspotSetupTypes'
+
+const HotspotSetupAddTxnScreen = () => {
+  const navigation = useNavigation<HotspotSetupNavigationProp>()
+  const { t } = useTranslation()
+  const [fee, setFee] = useState(0)
+  const [submitting, setSubmitting] = useState(false)
+  const { updateHotspotStatus, addGatewayTxn } = useConnectedHotspotContext()
+
+  const {
+    account: { account },
+    connectedHotspot: { mac, freeAddHotspot, status: hotspotStatus },
+  } = useSelector((state: RootState) => state)
+  const { hasLocationPermission } = usePermissionManager()
+
+  const navNext = async () => {
+    if (hotspotStatus === 'global') {
+      navigation.popToTop()
+      return
+    }
+
+    if (hotspotStatus === 'owned' || hotspotStatus === 'new') {
+      const hasLocPerm = await hasLocationPermission()
+      if (hasLocPerm) navigation.push('HotspotLocationFeeScreen')
+      else navigation.push('EnableLocationScreen')
+    }
+  }
+
+  const act = async () => {
+    if (hotspotStatus === 'new') {
+      setSubmitting(true)
+
+      const txn = await addGatewayTxn()
+      setSubmitting(false)
+      if (txn) {
+        navNext()
+      }
+    } else {
+      navNext()
+    }
+  }
+
```

Comment:
@tyler-whitman @allenan ↓↓ Can you check the logic here?

Reply:
I think for now our logic can be if it's "free" (ie it came back as registered in the staking server, so helium, inc pays the dc, making it free from the user's perspective), then show the 0 fee screen. If not, then show the hardcoded fee screen. We may want to even change what that fee screen shows in that case because even if you had enough DC to cover that yourself, you wouldn't be able to add it because it's not made by a helium-approved manufacturer. So as long as we just get the free vs not-free logic working for now, that's probably fine. We can revise a bit later